src/stores/pacbioLibraries.js

```diff
@@ -0,0 +1,122 @@
+            },
+          },
+        },
+      }
+      //TODO_LIBRARY_CHANGE:Needs to be changed to library
+      const request = rootState.api.traction.pacbio.libraries
+      const promise = request.create({ data: body, include: 'tube,primary_aliquot' })
+      const { success, data: { included = [] } = {}, errors } = await handleResponse(promise)
+      const { tubes: [tube = {}] = [] } = groupIncludedByResource(included)
+      const { attributes: { barcode = '' } = {} } = tube
+      return { success, barcode, errors }
+    },
+
+    /**Delete all libraries with given library ids */
+    async deleteLibraries(libraryIds) {
+      const rootStore = useRootStore()
+      const promises = rootStore.api.traction.pacbio.libraries.destroy(libraryIds)
+      //TODO_LIBRARY_CHANGE: Update all components (which calls deleteLibraries) to use the new response format
+      const responses = await Promise.all(promises.map((promise) => handleResponse(promise)))
+      return responses
+    },
+
+    /**Set libraries in store with those fetched using given filter and page  */
+    async setLibraries(filter, page) {
+      const rootStore = useRootStore()
+      const pacbioLibraries = rootStore.api.traction.pacbio.libraries
+      const promise = pacbioLibraries.get({
+        page,
+        filter,
+        include: 'request,tag,tube',
+      })
+      const response = await handleResponse(promise)
+
+      const { success, data: { data, included = [], meta = {} } = {}, errors = [] } = response
+      const { tubes, tags, requests } = groupIncludedByResource(included)
+
+      if (success) {
+        /* 
+      Here we build library objects to include necessary relational data
+      for the pacbio libraries page
+      */
+        const libraries = data.map((library) => {
+          return {
+            /** */
+            id: library.id,
+            ...library.attributes,
+            tag_group_id: tags?.find((tag) => tag.id == library.relationships.tag.data?.id)
+              ?.attributes.group_id,
+            sample_name: requests?.find(
+              (request) => request.id == library.relationships.request.data?.id,
+            )?.attributes.sample_name,
+            barcode: tubes?.find((tube) => tube.id == library.relationships.tube.data?.id)
+              ?.attributes.barcode,
+          }
+        })
+        this.libraries = libraries
+      }
+      return { success, errors, meta }
+    },
+
+    /**Fetch all tag sets */
+    async fetchPacbioTagSets() {
+      const rootStore = useRootStore()
+      const promise = rootStore.api.traction.pacbio.tag_sets.get({ include: 'tags' })
+      const response = await handleResponse(promise)
+      const { success, data: { data, included = [] } = {}, errors = [] } = response
+      if (success) {
+        const tagSets = dataToObjectById({ data, includeRelationships: true })
+        const tags = dataToObjectById({ data: included })
+        const tagSetChoices = []
```

Comment:
Can we update tagSetChoices and tagChoices directly instead of having a temporary variable now?

Reply:
Again I think we might be doing stuff we don't need to. It looks like we are setting up all of the tag sets when in essence most of the time they won't be using it. It would be good to work through it to see exactly what it going on and also look at what we are doing in the pools page.

Reply:
@stevieing It would be great if we could discuss a few of these comments tomorrow.